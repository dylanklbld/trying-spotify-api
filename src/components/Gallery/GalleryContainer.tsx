import React from 'react';
import {Spinner} from '../Spinner';
import styled from "styled-components";
const containersColor = `#57717D`

const AreaContainer = styled.div<{isLoading?:boolean}>`
    background: ${containersColor};
    display: flex;
    flex-wrap: wrap;
    padding: 5px 10px;
    justify-content: space-evenly;
    align-items:center;
    opacity: ${props => props.isLoading ? 0.1 : 1};
    z-index: 1;
    pointer-events:${props => props.isLoading ? 'none' : 'auto'};
`;

type Props = {
    isLoading?: boolean,
    children: React.ReactNodeArray
  }
  
  const GalleryContainer = ({children, isLoading}: Props): React.ReactElement => { 
    return (<AreaContainer isLoading={isLoading}>
        {isLoading && <Spinner width={100} height={100}/>}
        <>{children}</>
      </AreaContainer>)
  }

export default GalleryContainer;