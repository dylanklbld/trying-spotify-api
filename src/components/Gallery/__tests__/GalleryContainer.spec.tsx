import { fireEvent, render, screen } from '@testing-library/react';

import GalleryContainer from '../GalleryContainer';
import React from 'react';

jest.mock("../../Spinner", () => ({Spinner:(props:any) => <div data-testid="spinner" {...props} />}));

describe('GalleryContainer', () => {
  it('should render GalleryContainer with children', () => {
    const {container} = render(<GalleryContainer isLoading={false}><p>OUTPUT</p><p>OUTPUT</p></GalleryContainer>)
    expect(container).toMatchSnapshot()
  })

  it('should render GalleryContainer with children and spinner', () => {
    const {container} = render(<GalleryContainer isLoading={true}><p>OUTPUT</p><p>OUTPUT</p></GalleryContainer>)
    screen.getByTestId("spinner")
    expect(container).toMatchSnapshot()
  })

})