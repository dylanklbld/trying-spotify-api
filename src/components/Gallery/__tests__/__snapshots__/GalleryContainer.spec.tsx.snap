// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryContainer should render GalleryContainer with children 1`] = `
<div>
  <div
    class="sc-bdfBwQ dKNVSb"
  >
    <p>
      OUTPUT
    </p>
  </div>
</div>
`;

exports[`GalleryContainer should render GalleryContainer with children and spinner 1`] = `
<div>
  <div
    class="sc-bdfBwQ eczmek"
  >
    <div
      data-testid="spinner"
      height="100"
      width="100"
    />
    <p>
      OUTPUT
    </p>
  </div>
</div>
`;
